File: test09/distinct_lines_old.py
#!/usr/bin/env python3

import sys, re

n = int(sys.argv[1])
nDistinct = 0
nRead = 0
seen = {}

for line in sys.stdin:
   line = re.sub(r'\s', '', line.lower())

   if line not in seen:
      nDistinct += 1
      seen[line] = 1
   nRead += 1
   if nDistinct >= n:
      print("%d distinct lines seen after %d lines read." % (nDistinct, nRead))
      sys.exit(1)

print("End of input reached after %d lines read -  %d different lines not seen." % (nRead, n))

File: test07/distinct_lines_old.pl
#!/usr/bin/perl -w

$max = $ARGV[0];
$nDistinct = 0;
$nRead = 0;

while ($line = <STDIN>) {
   $line = lc $line;
   $line =~ s/\s//g;
   #print "$line\n";

   if (!$seen{$line}) {
      $seen{$line}++;
      $nDistinct++;
   }
   $nRead++;

   if ($nDistinct >= $max) {
      print "$nDistinct distinct lines seen after $nRead lines read.\n";
      exit;
   }
}

print "End of input reached after $nRead lines read - $max different lines not seen.\n";
